CardParts/src/Classes/CardPartsBottomSheetViewController.swift

```diff
@@ -0,0 +1,525 @@
+    
+    /// Whether or not to respond to dragging on the handle. Default is true.
+    public var shouldListenToHandleDrag: Bool = true
+    
+    /// Whether or not to respond to dragging in the content. Default is true.
+    public var shouldListenToContentDrag: Bool = true
+    
+    /// Whether or not to respond to dragging in the container. Default is true.
+    public var shouldListenToContainerDrag: Bool = true
+    
+    /// Whether or not to require a drag to start in the vertical direction. Default is true.
+    public var shouldRequireVerticalDrag: Bool = true
+    
+    /// Boolean value for whether or not bottom sheet should automatically add to its height to account for bottom safe area inset. Default is true.
+    public var adjustsForSafeAreaBottomInset: Bool = true
+    
+    /// Callback function to be called when bottom sheet is done preseting.
+    public var didShow: (() -> Void)? = nil
+    
+    /// Callback function to be called when bottom sheet is done dismissing itself.
+    /// - Parameter dismissalType: information about how the bottom sheet was dismissed
+    public var didDismiss: ((_ dismissalType: BottomSheetDismissalType) -> Void)? = nil
+    
+    /// Callback function to be called when bottom sheet height changes from dragging or a call to `updateHeight`.
+    public var didChangeHeight: ((_ newHeight: CGFloat) -> Void)? = nil
+    
+    /// Gesture recognizers that should block the vertical dragging of bottom sheet. Will automatically find and use all gesture recognizers if nil, otherwise will use recognizers in the array. Default is empty array.
+    public var preferredGestureRecognizers: [UIGestureRecognizer]? = []
+    
+    // MARK: Private variables
+    private var bottomSheetContainerVC: UIViewController = UIViewController()
+    private var bottomSheetHeight: CGFloat = 0
+    private var darkOverlay: UIView = UIView()
+    private var prevTouchHeight: CGFloat = 0
+    private var bottomSheetTopConstraint: NSLayoutConstraint!
+    private var viewTopConstraint: NSLayoutConstraint!
+    private var _contentHeight: CGFloat = 0
+    private var isShowingBottomSheet: Bool = false
+    
+    // MARK: Public functions
+    /// Adds the bottom sheet to the specified view, and makes calls to set up and display bottom sheet
+    /// - Parameter view: View on which to add the bottom sheet. Defaults to key window.
+    public func presentBottomSheet(on hostView: UIView? = UIApplication.shared.keyWindow) {
+        guard let hostView = hostView, !isShowingBottomSheet else { return }
+        self.view.translatesAutoresizingMaskIntoConstraints = false
+        
+        hostView.addSubview(self.view)
+        
+        NSLayoutConstraint.activate([
+            self.view.leadingAnchor.constraint(equalTo: hostView.leadingAnchor),
+            self.view.trailingAnchor.constraint(equalTo: hostView.trailingAnchor),
+            self.view.bottomAnchor.constraint(equalTo: hostView.bottomAnchor)
+        ])
```

Comment:
Try using layout from `LayoutProxy.swift`

Reply:
Good call, did this elsewhere but missed this one